src/compiler/scala/tools/nsc/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -491,18 +491,22 @@ abstract class BCodeSkelBuilder extends BCodeHelpers {
         case dd : DefDef =>
           val sym = dd.symbol
           if (needsStaticImplMethod(sym)) {
-            val staticDefDef = global.gen.mkStatic(dd, traitImplMethodName(sym), _.cloneSymbol)
-            val forwarderDefDef = {
-              val forwarderBody = Apply(global.gen.mkAttributedRef(staticDefDef.symbol), This(sym.owner).setType(sym.owner.typeConstructor) :: dd.vparamss.head.map(p => global.gen.mkAttributedIdent(p.symbol))).setType(sym.info.resultType)
-              // we don't want to the optimizer to inline the static method into the forwarder. Instead,
-              // the backend has a special case to transitively inline into a callsite of the forwarder
-              // when the forwarder itself is inlined.
-              forwarderBody.updateAttachment(NoInlineCallsiteAttachment)
-              deriveDefDef(dd)(_ => global.atPos(dd.pos)(forwarderBody))
-            }
-            genDefDef(staticDefDef)
-            if (!sym.isMixinConstructor)
+            if (sym.isMixinConstructor) {
+              val statified = global.gen.mkStatic(dd, sym.name, _.cloneSymbol)
+              genDefDef(statified)
+            } else {
+              val forwarderDefDef = {
+                val dd1 = global.gen.mkStatic(deriveDefDef(dd)(_ => EmptyTree), traitSuperAccessorName(sym), _.cloneSymbol)
+                dd1.symbol.setFlag(Flags.ARTIFACT).resetFlag(Flags.OVERRIDE)
+                deriveDefDef(dd1)(_ =>
+                  atPos(dd1.pos)(
+                    Apply(Select(global.gen.mkAttributedIdent(dd1.symbol.paramss.head.head).setType(sym.owner.typeConstructor), dd.symbol),
+                    dd.vparamss.head.map(p => global.gen.mkAttributedIdent(p.symbol))).updateAttachment(UseInvokeSpecial))
```

Comment:
It seems a bit hackish to refer to `dd`'s parameter symbols within the body of `dd1`. Should it be `dd1.vparamss.head.tail`?